### PR TITLE
[Misc] Weblate now ignore empty translation files

### PR DESCRIPTION
**Pull request by surli**

### CODE REVIEW

File: weblate-scripts/pre_commit.py [hunk 1]
```diff
@@ -43,13 +43,14 @@ def properties_to_xwiki_xml(file_path, path_prefix, lang):
...
         properties.load(f_properties.read())
 
-    title = properties.get_value("{}.title".format(file_name))
-    content = properties.get_value("{}.content".format(file_name))
-    xml_file = XmlFile()
-    xml_file.load(path_prefix + file_path)
-    xml_file.set_tag_content("title", title)
-    xml_file.set_tag_content("content", content)
-    xml_file.write(path_prefix + file_path)
+    if not properties.is_empty():
```
Comment (atallahade) [inline weblate-scripts/pre_commit.py:46]: Instead you can explicitly `return` the method and add a log message (a simple print, or something else if you have time).

File: weblate-scripts/pre_commit.py [hunk 2]
```diff
@@ -72,14 +73,16 @@ def properties_to_xwiki_xml_properties(file_path, path_prefix, base_file_name, l
     base_properties = PropertiesFile()
     base_properties.load(content)
 
-    # Replace keys with the current translation
-    base_properties.replace_with(properties)
-    base_properties.filter_export()
+    if not properties.is_empty():
```
Comment (atallahade) [inline weblate-scripts/pre_commit.py:71]: Same as above, and you will be able to do the check earlier.

File: weblate-scripts/pre_commit.py [hunk 3]
```diff
@@ -98,16 +101,18 @@ def properties_to_xwiki_properties(file_path, path_prefix, base_file_name, lang)
     with open(properties_path, "r") as f_properties:
         properties.load(f_properties.read())
 
-    # Use the base translation file as template
-    base_properties = PropertiesFile()
-    with open(path_prefix + base_file_name, "r") as f_properties:
-        base_properties.load(f_properties.read())
+    if not properties.is_empty():
```
Comment (atallahade) [inline weblate-scripts/pre_commit.py:107]: Same as above.

File: weblate-scripts/pre_commit.py (hunk 1 above)
Comment (surli) [reply weblate-scripts/pre_commit.py:46]: ok for adding a log in a else but I don't really understand what you mean by 
>  you can explicitly return the method

File: weblate-scripts/pre_commit.py (hunk 2 above)
Comment (surli) [reply weblate-scripts/pre_commit.py:71]: hm indeed I mixed up between base_properties and properties. I'll check it above

File: weblate-scripts/pre_commit.py (hunk 1 above)
Comment (atallahade) [reply weblate-scripts/pre_commit.py:46]:
```suggestion
    if properties.is_empty():
        return
```

Comment (surli) [reply weblate-scripts/pre_commit.py:46]: Hm. I prefer handling the not case here, I find it clearer.

**Comment by surli** [comment]
@atallahade for me it's ready

File: weblate-scripts/common.py [hunk 4]
```diff
@@ -234,6 +234,13 @@ def map_properties(self):
                     print "Warning: {} already exists.".format(key)
                 self.properties[key] = value
 
+    def is_empty(self):
+        """Returns true if the all properties are empty"""
+        if not self.properties or len("".join(self.properties.values())) == 0:
```
Comment (atallahade) [inline weblate-scripts/common.py]:
```suggestion
        return not self.properties or len("".join(self.properties.values())) == 0:
```

**Review by atallahade — APPROVED** [review]

File: weblate-scripts/common.py (hunk 4 above)
Comment (surli) [reply weblate-scripts/common.py]: indeed...

**Pull request merged**